include/powerloader/mirror.hpp

```diff
@@ -62,11 +62,29 @@ namespace powerloader
         }
     };
 
+    inline std::string strip_trailing_slash(const std::string& s)
+    {
+        if (s.size() > 0 && s.back() == '/' && s != "file://")
+        {
+            return s.substr(0, s.size() - 1);
+        }
+        return s;
+    }
+
     // mirrors should be dict -> urls mapping
-    struct POWERLOADER_API Mirror
+    class POWERLOADER_API Mirror
     {
-        Mirror(MirrorID id, const Context& ctx, const std::string& url);
-        Mirror(const Context& ctx, const std::string& url);
+    public:
+        Mirror(const MirrorID& id, const Context& ctx, const std::string& url)
```

Comment:
`id` can be passed by value and moved in for optimal performance (the move is forced to be no-op - technique known as "sink value"). But thats minor here.